airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -53,68 +79,71 @@
   private static final int MIN_RECORDS = 500;
   private static final int BATCH_SIZE = 500;
 
-  private final BufferedWriteOperations destination;
-  private Map<String, DestinationWriteContext> writeConfigs;
-  private Map<String, CloseableQueue<byte[]>> writeBuffers;
+  private final VoidCallable onStart;
+  private final CheckedBiConsumer<String, Stream<AirbyteRecordMessage>, Exception> recordWriter;
+  private final CheckedConsumer<Boolean, Exception> onClose;
+  private final Set<String> streamNames;
+  private final Map<String, CloseableQueue<byte[]>> writeBuffers;
   private final ScheduledExecutorService writerPool;
   private final ConfiguredAirbyteCatalog catalog;
 
-  public BufferedStreamConsumer(BufferedWriteOperations destination, ConfiguredAirbyteCatalog catalog) {
-    this.destination = destination;
-    this.writeConfigs = new HashMap<>();
-    this.writeBuffers = new HashMap<>();
-    this.writerPool = Executors.newSingleThreadScheduledExecutor();
+  private boolean hasStarted;
+
+  public BufferedStreamConsumer(VoidCallable onStart,
```

Comment:
note: this consumer is now totally ignorant of write configs and jdbc and can theoretically be reused by any destination that wants to leverage the on-disk queue.